src/lib/state/DY.Lib.State.Typed.fst

```diff
@@ -5,8 +5,22 @@ open DY.Core
 open DY.Lib.Comparse.Glue
 open DY.Lib.State.Tagged
 
+class local_state (a:Type0) = {
+  tag: string;
+  [@@@FStar.Tactics.Typeclasses.tcinstance]
+  format: parseable_serializeable bytes a;
+}
+
+val mk_local_state_instance:
+  #a:Type0 -> {|parseable_serializeable bytes a|} -> string ->
+  local_state a
+let mk_local_state_instance #a #format tag = {
+  tag;
+  format;
+}
+
 noeq
-type typed_session_pred {|crypto_invariants|} (a:Type) {|parseable_serializeable bytes a|} = {
+type local_state_predicate {|crypto_invariants|} (a:Type) {|parseable_serializeable bytes a|} = {
```

Comment:
Should we also rename the [`state_predicate` in Core](https://github.com/REPROSEC/dolev-yao-star-extrinsic/blob/6fe7c9d40a097a5bdb2622fe95a0ebf8e9df3f75/src/core/DY.Core.Trace.Invariant.fst#L31) to contain `global` to make the different levels of the predicates even more clear?

Reply:
Arguably the whole zoo of predicates would benefit some renaming, I will open an issue after merging.